pkg/statistics/region.go

```diff
@@ -15,46 +15,80 @@
-	StorePeerCount   map[uint64]int   `json:"store_peer_count"`
-	StoreLeaderSize  map[uint64]int64 `json:"store_leader_size"`
-	StoreLeaderKeys  map[uint64]int64 `json:"store_leader_keys"`
-	StorePeerSize    map[uint64]int64 `json:"store_peer_size"`
-	StorePeerKeys    map[uint64]int64 `json:"store_peer_keys"`
+	Count                int               `json:"count"`
+	EmptyCount           int               `json:"empty_count"`
+	StorageSize          int64             `json:"storage_size"`
+	UserStorageSize      int64             `json:"user_storage_size"`
+	StorageKeys          int64             `json:"storage_keys"`
+	StoreLeaderCount     map[uint64]int    `json:"store_leader_count"`
+	StorePeerCount       map[uint64]int    `json:"store_peer_count"`
+	StoreLeaderSize      map[uint64]int64  `json:"store_leader_size"`
+	StoreLeaderKeys      map[uint64]int64  `json:"store_leader_keys"`
+	StorePeerSize        map[uint64]int64  `json:"store_peer_size"`
+	StorePeerKeys        map[uint64]int64  `json:"store_peer_keys,omitempty"`
+	StoreWriteBytes      map[uint64]uint64 `json:"store_write_bytes,omitempty"`
+	StoreWriteKeys       map[uint64]uint64 `json:"store_write_keys,omitempty"`
+	StoreWriteQuery      map[uint64]uint64 `json:"store_write_query,omitempty"`
+	StoreLeaderReadBytes map[uint64]uint64 `json:"store_leader_read_bytes,omitempty"`
+	StoreLeaderReadKeys  map[uint64]uint64 `json:"store_leader_read_keys,omitempty"`
+	StoreLeaderReadQuery map[uint64]uint64 `json:"store_leader_read_query,omitempty"`
+	StorePeerReadBytes   map[uint64]uint64 `json:"store_peer_read_bytes,omitempty"`
+	StorePeerReadKeys    map[uint64]uint64 `json:"store_peer_read_keys,omitempty"`
+	StorePeerReadQuery   map[uint64]uint64 `json:"store_peer_read_query,omitempty"`
+	StoreEngine          map[uint64]string `json:"store_engine,omitempty"`
+}
+
+// GetRegionStatsOption is used to filter the peer statistics.
+type GetRegionStatsOption func(peer *metapb.Peer) bool
+
+// WithStoreMapOption returns a GetRegionStatsOption that filters the peer by store map.
+func WithStoreMapOption(storeMap map[uint64]string) GetRegionStatsOption {
+	return func(peer *metapb.Peer) bool {
+		_, exist := storeMap[peer.GetStoreId()]
+		return exist
+	}
 }
 
 // GetRegionStats sums regions' statistics.
-func GetRegionStats(regions []*core.RegionInfo) *RegionStats {
+func GetRegionStats(regions []*core.RegionInfo, cluster RegionStatInformer, opts ...GetRegionStatsOption) *RegionStats {
 	stats := newRegionStats()
 	for _, region := range regions {
-		stats.Observe(region)
+		stats.Observe(region, cluster, opts...)
 	}
 	return stats
 }
 
 func newRegionStats() *RegionStats {
 	return &RegionStats{
-		StoreLeaderCount: make(map[uint64]int),
-		StorePeerCount:   make(map[uint64]int),
-		StoreLeaderSize:  make(map[uint64]int64),
-		StoreLeaderKeys:  make(map[uint64]int64),
-		StorePeerSize:    make(map[uint64]int64),
-		StorePeerKeys:    make(map[uint64]int64),
+		StoreLeaderCount:     make(map[uint64]int),
```

Comment:
Is there any OOM risk?

Reply:
The length of the map is mostly small(less than 200). The API is primarily used by DBAs with a low frequent usage rate.